### PR TITLE
move default solver tests to OrdinaryDiffEqDefault

### PR DESCRIPTION
**Pull request by oscardssmith**

### CODE REVIEW

File: lib/OrdinaryDiffEqDefault/Project.toml [hunk 1]
```diff
@@ -16,6 +16,7 @@ OrdinaryDiffEqVerner = "79d7bb75-1356-48c1-b8c0-6832512096c2"
 PrecompileTools = "aea7be01-6a6a-4083-8856-8a6e6704d82a"
 Preferences = "21216c6a-2e73-6563-6e65-726566657250"
 Reexport = "189a3867-3050-52da-a836-e630ba90ab69"
+StaticArrays = "90137ffa-7385-5640-81b9-e52037218182"
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqDefault/Project.toml:19]: why is this needed as a dep?

Comment (oscardssmith) [reply lib/OrdinaryDiffEqDefault/Project.toml:19]: it could be test only, but since Rosenbrock depends on it explicitly, it already is a dep.

Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqDefault/Project.toml:19]:
```suggestion
```

File: lib/OrdinaryDiffEqDefault/Project.toml [hunk 2]
```diff
@@ -33,14 +34,16 @@ Preferences = "1.4.3"
...
+StaticArrays = "1.0"
 Test = "<0.0.1, 1"
 julia = "1.10"
 
 [extras]
 DiffEqDevTools = "f3b72e0c-5b89-59e1-b016-84e28bfd966d"
 Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
 SafeTestsets = "1bc83da4-3b8d-516f-aca4-4fe02f6d838f"
+SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqDefault/Project.toml:46]:
```suggestion
SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
StaticArrays = "90137ffa-7385-5640-81b9-e52037218182"
Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
```
